driver-core/src/main/com/mongodb/MongoCredential.java

```diff
@@ -138,19 +148,20 @@
 
     /**
      * Creates a MongoCredential instance with an unspecified mechanism.  The client will negotiate the best mechanism based on the
-     * version of the server that the client is authenticating to.  If the server version is 3.0 or higher,
-     * the driver will authenticate using the SCRAM-SHA-1 mechanism.  Otherwise, the driver will authenticate using the MONGODB_CR
-     * mechanism.
+     * version of the server that the client is authenticating to.
      *
+     * <p>If the server version is 4.0 or higher, the driver will negotiate with the server preferring the SCRAM-SHA-256 mechanism. 3.x
```

Comment:
Not sure what it means practically to "prefer the SCRAM-SHA-256 mechanism".

Reply:
It depends on the configured user mechanisms, where SHA-256 is preferred. As its nuanced it best left to the official documentation to cover.